src/displayBackend/wayland/Surface.cpp

```diff
@@ -99,7 +113,17 @@ void Surface::draw(FrameBufferPtr frameBuffer,
 
 	wl_surface_commit(mWlSurface);
 
+#ifdef WITH_WAYLAND_PRESENTATION_API
+	auto presentation = mCompositor->getPresentation();
+	if (presentation)
+	{
+		mFeedback = wp_presentation_feedback(presentation, mWlSurface);
+		wp_presentation_feedback_add_listener(mFeedback, &mPresentationFeedbackListener, this);
```

Comment:
Do you need to remove listener during object destruction?

Reply:
No need to, since after the presentation will be destructed, there will be no callbacks.